### ISSUE TITLE
Extract fitted models

### ISSUE THREAD

**Issue by agrudkow** [LudvigOlsen/cvms#23]
Is there a way to extract fitted models? I would like to test my model on data which it haven't seen to get true metrics and I cannot find a way to do it.

**Comment by LudvigOlsen** [comment]
Hi @agrudkow,

Currently, this is not supported no. The models haven't seen the data they are tested on during cross-validation though, but a testing on a hold-out test set is a great approach.

Normally, I would probably train the final model on the full training data and test that on the hold-out test set (e.g. with `validate()` or `validate_fn()`). Would that work for you? Or do you prefer to check each of the fitted CV models on the hold-out test set?

In the future, I might add an option to save the model objects to disk during training. With the potential of 1000s of fitted models in repeated cross-validation, keeping them in-memory during training might not be feasible. If you can wait a month, I can add this to my ToDo for June?

Alternatively, if your model is fairly simple and the coefficients were saved in the results (in the `Coefficients` column; not all model types are supported), you might be able to use those. For more advanced models that could be a hassle though.

Best,
Ludvig

**Comment by agrudkow** [comment]
Actually, now when I read your response I understood that I misguided myself into wrong perception about cross validation. You are right that after training several models with help of cross-validation we choose the best one and train it on the whole dataset and evaluate its final performance on test set. As a reason of that there is no need to extract models from cross validation (at least not in my use case). Thanks for explanation (maybe not direct, but it helped).

Best regards,
agrudkow

**Comment by LudvigOlsen** [comment]
Good to hear :)

I have encountered papers where an ensemble of the CV models were tested. So it might be worth adding as an option. But I haven't needed myself either.

Questions and suggestions are always highly welcome :)
